docs/using-connect/integration.mdx

```diff
@@ -0,0 +1,374 @@
+
+## Requirements
+- Go 1.22+
+- Cosmos SDK v0.50+
+
+
+
+## Integrating Connect
+
+Integrating Connect into your Cosmos SDK application requires a few simple steps.
+
+<Steps>
+    <Step title="Add Keepers">
+        Add the `x/marketmap` and `x/oracle` keepers to your app.
+    </Step>
+    <Step title="Setup Oracle Client">
+        Set up the Oracle client the application will use to get prices from the Connect oracle.
+    </Step>
+    <Step title="Set ABCI Method">
+        Set the PreBlock ABCI method, which is responsible for aggregating prices and writing them to the application's state.
+    </Step>
+    <Step title="Configure Vote Extensions">
+        Configure vote extensions with compression and storage strategies.
+    </Step>
+</Steps>
+
+## Keepers
+
+First, add the [x/marketmap](https://github.com/skip-mev/slinky/blob/main/x/marketmap/README.md) and [x/oracle](https://github.com/skip-mev/slinky/tree/main/x/oracle) keepers to the Application.
+
+```go app.go
+package app
+
+import (
+    // ... other imports
+
+	marketmapkeeper "github.com/skip-mev/slinky/x/marketmap/keeper"
+	oraclekeeper "github.com/skip-mev/slinky/x/oracle/keeper"
+)
+
+type App struct {
+	// ... other fields
+
+	OracleKeeper     *oraclekeeper.Keeper
+	MarketMapKeeper  *marketmapkeeper.Keeper
+}
+```
+
+Then, add them to the dependency injection system.
+
+```go app.go
+	err := depinject.Inject(
+		// ... other arguments
+		&app.MarketMapKeeper,
+		&app.OracleKeeper,
+	)
+```
+
+Finally, once the app is built with the `appBuilder`, finish the initialization of the `MarketMapKeeper` by setting the hooks.
+
+```go app.go
+	app.App = appBuilder.Build(db, traceStore, baseAppOptions...)
+
+	app.MarketMapKeeper.SetHooks(app.OracleKeeper.Hooks())
+```
+
+## Oracle Client
+
+Create a method to construct and return the oracle client and metrics.
+
+```go oracle.go
+package app
+
+import (
+    "github.com/cosmos/cosmos-sdk/server/types"
+    oracleconfig "github.com/skip-mev/slinky/oracle/config"
+    oracleclient "github.com/skip-mev/slinky/service/clients/oracle"
+    servicemetrics "github.com/skip-mev/slinky/service/metrics"
+)
+
+// initializeOracle initializes the oracle client and metrics.
+func (app *App) initializeOracle(appOpts types.AppOptions) (oracleclient.OracleClient, servicemetrics.Metrics, error) {
+	// Read general config from app-opts, and construct oracle service.
+	cfg, err := oracleconfig.ReadConfigFromAppOpts(appOpts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// If app level instrumentation is enabled, then wrap the oracle service with a metrics client
+	// to get metrics on the oracle service (for ABCI++). This will allow the instrumentation to track
+	// latency in VerifyVoteExtension requests and more.
+	oracleMetrics, err := servicemetrics.NewMetricsFromConfig(cfg, app.ChainID())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Create the oracle service.
+	oracleClient, err := oracleclient.NewClientFromConfig(
```

Comment:
Would update this with `NewPriceDaemonClientFromConfig`

Reply:
done [use NewPriceDaemon](https://github.com/skip-mev/slinky/pull/618/commits/4f5097b8b194b190796178e7eeaeb596fb622dd5)